source/WeightedList.h

```diff
@@ -51,13 +55,13 @@ class WeightedList {
 	const_iterator end() const noexcept { return choices.end(); }
 
 	void clear() noexcept { choices.clear(); total = 0; }
-	std::size_t size() const noexcept { return choices.size(); }
+	int size() const noexcept { return choices.size(); }
```

Comment:
it's fine to store the weights as ints but `size()` / `TotalWeight()`should always be unsigned

Reply:
Well now that we're capturing weight as an int, I think it'd be fine to store weights as size_ts again, as the only issue with storing weights as a size_t was negative inputs.